#!/usr/bin/env python
PACKAGE = "safety_limiter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("freq", double_t, 0, "", 6.0, 0.0, 100.0)
gen.add("cloud_timeout", double_t, 0, "", 0.8, 0.0, 10.0)
gen.add("disable_timeout", double_t, 0, "", 0.1, 0.0, 10.0)
gen.add("lin_vel", double_t, 0, "", 0.5, 0.0, 10.0)
gen.add("lin_acc", double_t, 0, "", 1.0, 0.0, 10.0)
gen.add("ang_vel", double_t, 0, "", 0.8, 0.0, 10.0)
gen.add("ang_acc", double_t, 0, "", 1.6, 0.0, 10.0)
gen.add("z_range_min", double_t, 0, "", 0.0, -3.0, 3.0)
gen.add("z_range_max", double_t, 0, "", 0.5, -3.0, 3.0)
gen.add("dt", double_t, 0, "", 0.1, 0.0, 1.0)
gen.add("d_margin", double_t, 0, "", 0.2, 0.0, 1.0)
gen.add("d_escape", double_t, 0, "", 0.05, 0.0, 1.0)
gen.add("yaw_margin", double_t, 0, "", 0.2, 0.0, 1.57)
gen.add("yaw_escape", double_t, 0, "", 0.05, 0.0, 1.57)
gen.add("downsample_grid", double_t, 0, "", 0.05, 0.0, 1.0)
gen.add("hold", double_t, 0, "", 0.0, 0.0, 10.0)
gen.add("allow_empty_cloud", bool_t, 0, "", False)

exit(gen.generate(PACKAGE, "safety_limiter", "SafetyLimiter"))
